sdk/src/main/java/com/kin/ecosystem/transfer/presenter/AccountInfoPresenter.java

```diff
@@ -0,0 +1,146 @@
+import android.support.annotation.NonNull;
+
+import com.kin.ecosystem.base.BasePresenter;
+import com.kin.ecosystem.common.exception.BlockchainException;
+import com.kin.ecosystem.core.data.blockchain.BlockchainSourceImpl;
+import com.kin.ecosystem.transfer.AccountInfoManager;
+import com.kin.ecosystem.transfer.view.IAccountInfoView;
+
+public class AccountInfoPresenter extends BasePresenter<IAccountInfoView> implements IAccountInfoPresenter {
+    private static final String EXTRA_SOURCE_APP_NAME = "EXTRA_SOURCE_APP_NAME";
+
+    private AccountInfoManager accountInfoManager;
+    private AccountInfoAsyncTask asyncTask;
+    private boolean taskComplete = false;
+    private boolean completeWithSuceess = false;
+    private boolean isPaused = false;
+
+    public AccountInfoPresenter(@NonNull AccountInfoManager accountInfoManager, @NonNull IAccountInfoView view, @NonNull Intent intent) {
+        this.accountInfoManager = accountInfoManager;
+        processIntent(view, intent);
+        view.attachPresenter(this);
+        startInit();
+    }
+
+    private void processIntent(IAccountInfoView view, Intent intent) {
+        if (intent != null && intent.hasExtra(EXTRA_SOURCE_APP_NAME)) {
+            String sourceApp = intent.getStringExtra(EXTRA_SOURCE_APP_NAME);
+            if (!sourceApp.isEmpty()) {
+                view.updateSourceApp(sourceApp);
+            } else {
+                onError(view);
+            }
+        } else {
+            onError(view);
+        }
+    }
+
+    @Override
+    public void agreeClicked() {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondOk();
+            if (view != null) {
+                view.close();
+            }
+        }
+    }
+
+    @Override
+    public void backButtonPressed() {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondCancel();
+        }
+    }
+
+    @Override
+    public void closeClicked() {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondCancel();
+            if (view != null) {
+                view.close();
+            }
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        asyncTask.cancel(true);
+        asyncTask = null;
+        super.onDetach();
+    }
+
+    @Override
+    public void onResume() {
+        isPaused = false;
+        if (taskComplete) {
+            onInitComplete(completeWithSuceess);
+        }
+    }
+
+    @Override
+    public void onPause() {
+        isPaused = true;
+    }
+
+    private void startInit() {
+        asyncTask = new AccountInfoAsyncTask();
+        asyncTask.execute();
+    }
+
+    private void onError(IAccountInfoView view) {
+        if (accountInfoManager != null) {
+            accountInfoManager.respondError();
+            if (view != null) {
+                view.close();
+            }
+        }
+    }
+
+    private void onTaskComplete(Boolean success) {
+        if (!isPaused) {
```

Comment:
I don't think we need isPaused, taskComplete and completeWithSuceess.
onTaskComplete can be called even if Activity is paused.
We only need to protect against calling onTaskComplete after Activity was destroyed

Reply:
no cause after task complete - it change in the ui the enable of the button
i dont want to call it when we are on the background